cmd/segment-reaper/observer.go

```diff
@@ -143,11 +145,88 @@ func (obsvr *observer) processSegment(ctx context.Context, path metainfo.ScopedP
 // analyzeProject analyzes the objects in obsv.objects field for detecting bad
 // segments and writing them to objs.writer.
 func (obsvr *observer) analyzeProject(ctx context.Context) error {
-	// TODO this part will be implemented in next PR
-	// TODO(if): For what is this?
+	for bucket, objects := range obsvr.objects {
+		for path, object := range objects {
+			if object.skip {
+				continue
+			}
+
+			brokenObject := false
+			if object.hasLastSegment {
+				if object.expectedNumberOfSegments == 0 {
+					if !object.segments.IsSequence() {
+						brokenObject = true
+					}
+				} else if object.segments.Count() != int(object.expectedNumberOfSegments)-1 {
+					// TODO should we also check if its valid sequence
```

Comment:
I think that we can remove this `TODO`, I don't see the point of checking the sequence if we already detect that the object is broken.

Reply:
removed